tests/integration/devfile/cmd_devfile_registry_test.go

```diff
@@ -81,6 +83,30 @@ var _ = Describe("odo devfile registry command tests", func() {
 			helper.CmdShouldPass("odo", "registry", "delete", registryName, "-f")
 			helper.CmdShouldFail("odo", "create", "java-maven", "--registry", registryName)
 		})
+	})
 
+	Context("when working with git based registries", func() {
+		It("should show deprecation warning when when the git based registry is used", func() {
+			deprecated := "Deprecated"
+			docLink := "https://github.com/openshift/odo/tree/main/docs/public/git-registry-deprecation.adoc"
+			outstr, errstr := helper.CmdShouldPassIncludeErrStream("odo", "registry", "add", "RegistryFromGitHub", "https://github.com/odo-devfiles/registry")
+			co := fmt.Sprintln(outstr, errstr)
+			helper.MatchAllInOutput(co, []string{deprecated, docLink})
+			outstr, errstr = helper.CmdShouldPassIncludeErrStream("odo", "registry", "list")
+			co = fmt.Sprintln(outstr, errstr)
+			helper.MatchAllInOutput(co, []string{deprecated, docLink})
+			outstr, errstr = helper.CmdShouldPassIncludeErrStream("odo", "create", "nodejs", "--registry", "RegistryFromGitHub")
+			co = fmt.Sprintln(outstr, errstr)
+			helper.MatchAllInOutput(co, []string{deprecated, docLink})
+		})
+		It("should not show deprecation warning if git based registry is not used", func() {
```

Comment:
I think `if non-git-based registry is used" makes more sense than "if git-based registry is not used".

Reply:
Updating, will push shortly